### PR TITLE
irods_extractor_config

### PR DESCRIPTION
**Pull request by flass**
to accommodate process-level config for shared workflows

### CODE REVIEW

**Comment by flass** [comment]
not sure about maxForks rate limiting directive as might be redundant with seqirods resource request ?

File: configs/nextflow.config [hunk 1]
```diff
@@ -219,6 +219,36 @@ process {
...
+    }
+}
+
+subworkflows {
+
+    // Config for subworkflows from https://gitlab.internal.sanger.ac.uk/sanger-pathogens/pipelines/assorted-sub-workflows
+    // for IRODS_EXTRACTOR subworkflow
+    
+    withName:IRODS_QUERY {
+        maxForks = 1
```
Comment (Lfulcrum) [inline configs/nextflow.config:250]: I would be inclined to put `maxForks = 1` at the process level for the `BATON` process as that's the only one doing the iRODS querying in the workflow.

**Review by Lfulcrum — COMMENTED** [review]
It's a nice idea. Not sure if this becomes too specific for this config. It might be cleaner if we were to put it in a separate config file and inherit it in whatever pipeline needs it?

If a pipeline doesn't use a process that is referenced by the `withName` selector, does it just ignore the config or does it complain with some warning message? I can't remember...

File: configs/nextflow.config (hunk 1 above)
Comment (flass) [reply configs/nextflow.config]: agreed; done

**Comment by flass** [comment]
> It's a nice idea. Not sure if this becomes too specific for this config. It might be cleaner if we were to put it in a separate config file and inherit it in whatever pipeline needs it?
> 
> If a pipeline doesn't use a process that is referenced by the `withName` selector, does it just ignore the config or does it complain with some warning message? I can't remember...

ideally we could keep the config within the assorted-sub-workflow repo, alongside the subworflow def - making use of the config hints e.g. https://gitlab.internal.sanger.ac.uk/sanger-pathogens/pipelines/assorted-sub-workflows/-/blob/main/irods_extractor/subworkflows/irods.config. 
ideally we need to find a way to make that be included in the common config depending on the use of the corresponding workflow in the piepline - so that nothing has to be done for each piepline.  
another simpler way would be to add an `includeConfig(./assorted-sub-worflows//irods_extractor/subworkflows/irods.config)` statement in the importing pipeline's config - it has to be done for each piepline, but it remains a minimal one-liner...

**Comment by flass** [comment]
this is redundant with https://gitlab.internal.sanger.ac.uk/sanger-pathogens/pipelines/assorted-sub-workflows/-/merge_requests/16 and https://gitlab.internal.sanger.ac.uk/sanger-pathogens/pipelines/rvi-pipeline/-/merge_requests/12#note_214752 so dropping this

**Pull request closed without merging**